test/repositories/ITSAStatusRepositorySupportSpec.scala

```diff
@@ -23,35 +23,46 @@ import testUtils.UnitSpec
 
 class ITSAStatusRepositorySupportSpec extends UnitSpec with Matchers {
 
-  private val expectedToString = Map(NoStatus -> "U", Voluntary -> "V", Annual -> "A", Mandated -> "M")
+  private val expectedToString = Map(Voluntary -> "V", Annual -> "A", Mandated -> "M")
 
   "ITSAStatusRepositorySupport.statusToString" should {
     "convert to String" in {
-      statusToString(NoStatus) shouldBe expectedToString(NoStatus)
-      statusToString(Voluntary) shouldBe expectedToString(Voluntary)
-      statusToString(Annual) shouldBe expectedToString(Annual)
-      statusToString(Mandated) shouldBe expectedToString(Mandated)
+      statusToString(status = Voluntary) shouldBe expectedToString(Voluntary)
+      statusToString(status = Annual) shouldBe expectedToString(Annual)
+      statusToString(status = Mandated) shouldBe expectedToString(Mandated)
     }
 
-    "throw error for Dormant" in {
-      assertThrows[RuntimeException] {
-        statusToString(Dormant)
+    "allow NoStatus only for CY+1" in {
+      statusToString(status = NoStatus) shouldBe "U"
+    }
+
+    "throw error for other unsupported statuses" in {
+      Seq(Exempt, DigitallyExempt, Dormant).foreach { unsupportedStatus =>
+        assertThrows[RuntimeException] {
+          statusToString(status = unsupportedStatus)
+        }
```

Comment:
This test fails currently

`Expected exception java.lang.RuntimeException to be thrown, but no exception was thrown`